hack/verify-codegen.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE}")/..
+DIFFROOT="${SCRIPT_ROOT}/pkg"
+TMP_DIFFROOT="${SCRIPT_ROOT}/_tmp/pkg"
+_tmp="${SCRIPT_ROOT}/_tmp"
+
+cleanup() {
+    rm -rf "${_tmp}"
+}
+trap "cleanup" EXIT SIGINT
+
+cleanup
+
+mkdir -p "${TMP_DIFFROOT}"
+cp -a "${DIFFROOT}"/* "${TMP_DIFFROOT}"
+
+"${SCRIPT_ROOT}/hack/update-codegen.sh"
+echo "diffing ${DIFFROOT} against freshly generated codegen"
+ret=0
+diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
+cp -a "${TMP_DIFFROOT}"/* "${DIFFROOT}"
```

Comment:
There is no guarantee this restores `DIFFROOT` to its original state.  E.g. it's quite possible the `update-codegen` phase generated new extra files that will not be removed by this `cp`.

It would be safer to rename `DIFFROOT` to `TMP_DIFFROOT`, run `update-codegen`, `diff`, delete `DIFFROOT` and rename `TMP_DIFFROOT` to `DIFFROOT` (note the cleanup function needs to change in this case).

Alternatively, coax `update-codegen` to generate the files elsewhere and diff DIFFROOT to that new location (which you can safely delete on error / whatever).  This is the most reliable and simple method (assuming it is not too difficult to get `update-codegen` to generate the files in an alternative location (e.g. a temp directory created with mktemp).

Reply:
To be honest this file is the same taken from kube/openshift repos... I would not invest too much time on it.

Reply:
Leaves much to be desired, however OK then... ;-)